pkg/config/namewatcher.go

```diff
@@ -0,0 +1,174 @@
+	if err == nil {
+		rn.updateSecretNames(sNames.Data)
+	}
+	cmNames, err = client.GetClient().CoreV1().ConfigMaps(client.DefaultNamespace).Get(ctx, sharedConfigMapReservedNamesConfigMapName, metav1.GetOptions{})
+	if err == nil {
+		rn.updateConfigMapNames(cmNames.Data)
+	}
+	//go rn.watchReservedNamesConfigMap(ctx, stopCh, rn.sharedConfigMapReservedNamesWatch)
+	//go rn.watchReservedNamesConfigMap(ctx, stopCh, rn.sharedSecretReservedNamesWatch)
+	return rn //, nil
+}
+
+func (rn *ReservedNames) liveCheck() (map[string]string, map[string]string) {
+	rn.lock.Lock()
+	defer rn.lock.Unlock()
+	var secretMapCopy map[string]string
+	var configMapMapCopy map[string]string
+	if rn.currentSecretNames == nil {
+		cm, err := client.GetClient().CoreV1().ConfigMaps(client.DefaultNamespace).Get(context.Background(), sharedSecretReservedNamesConfigMapName, metav1.GetOptions{})
+		if err == nil && cm != nil && cm.Data != nil {
+			secretMapCopy = make(map[string]string)
+			for k, v := range cm.Data {
+				secretMapCopy[k] = v
+			}
+		}
+	}
+	if rn.currentConfigMapNames == nil {
+		cm, err := client.GetClient().CoreV1().ConfigMaps(client.DefaultNamespace).Get(context.Background(), sharedConfigMapReservedNamesConfigMapName, metav1.GetOptions{})
+		if err == nil {
+			configMapMapCopy := make(map[string]string)
+			for k, v := range cm.Data {
+				configMapMapCopy[k] = v
+			}
+		}
+	}
+	return secretMapCopy, configMapMapCopy
+}
+
+func (rn *ReservedNames) ValidateSharedSecretOpenShiftName(shareName, refNamespace, refName string) bool {
+	secretMapCopy, _ := rn.liveCheck()
+	if secretMapCopy != nil {
+		rn.updateSecretNames(secretMapCopy)
+	}
+	rn.lock.Lock()
+	defer rn.lock.Unlock()
+	v, ok := rn.currentSecretNames[shareName]
+	return innerValidate(shareName, v, refNamespace, refName, ok)
+}
+
+func (rn *ReservedNames) ValidateSharedConfigMapOpenShiftName(shareName, refNamespace, refName string) bool {
+	_, configMapCopy := rn.liveCheck()
+	if configMapCopy != nil {
+		rn.updateConfigMapNames(configMapCopy)
+	}
+	rn.lock.Lock()
+	defer rn.lock.Unlock()
+	v, ok := rn.currentConfigMapNames[shareName]
+	return innerValidate(shareName, v, refNamespace, refName, ok)
+}
+
+func startsWithOpenShift(shareName string) bool {
+	if strings.HasPrefix(shareName, "openshift-") ||
+		strings.HasPrefix(shareName, "openshift_") {
+		return true
+	}
+	return false
+}
+
+func innerValidate(shareName, v, refNamespace, refName string, ok bool) bool {
+	if !startsWithOpenShift(shareName) {
+		return true
+	}
+	if !ok {
+		return false
+	}
+	a := strings.Split(v, ":")
+	if len(a) < 2 {
+		//TODO log/event
+		return false
+	}
+	if strings.TrimSpace(a[0]) != strings.TrimSpace(refNamespace) ||
+		strings.TrimSpace(a[1]) != strings.TrimSpace(refName) {
+		return false
+	}
+	return true
+
+}
+
+//func (rn *ReservedNames) watchReservedNamesConfigMap(ctx context.Context, stopCh <-chan struct{}, w watch.Interface) {
```

Comment:
I'll clean this up once e2e's look OK. Don't think I need it but wanted to wait and make sure.